pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -309,25 +309,50 @@ func (state PipelineRunState) getNextTasks(candidateTasks sets.String) []*Resolv
 
 // getRetryableTasks returns a list of tasks which should be executed next when the pipelinerun is stopping, i.e.
 // a list of cancelled/failed tasks from candidateTasks which haven't exhausted their retries
+// If the PipelineTask has a Matrix, isCancelled returns true if any run is cancelled and all other runs are done.
```

Comment:
sorry to keep pushing on this-- this reads more like a docstring for `isCancelled` which makes me wonder if there is some overlap between those two functions?

To clarify, when the pipelinerun is in stopping mode, for a non-matrixed taskrun it will only schedule it if it's cancelled/failed and has remaining retries. For a matrixed taskrun, it will schedule a taskrun if it is cancelled or failed with remaining retries **only if all other matrixed taskruns are done** (which includes being cancelled)?

If any matrixed taskrun is cancelled, does that mean all the others will be cancelled? It seems to me like we should be scheduling a matrixed taskrun in stopping mode if it's cancelled/failed with remaining retries, regardless of the state of the other matrixed taskruns.

Reply:
You were right, switched it to using `t.isCancelled()` instead of repeating the same logic.

The docstring wasn't reflecting what was happening, even for non-matrixed cases.  This function returns a list of pipelinetasks which should be executed next when the pipelinerun is stopping i.e. a list of failed pipelinetasks from candidateTasks which haven't exhausted their retries. If a pipelinetask is cancelled, the retries are not exhausted - they are not retryable.

For matrixed cases, isCancelled returns true if any run is cancelled and all other runs are done. Cancelling one taskrun from a matrix doesn't cancel the others, just the same way that one failing doesn't fail the others.

Reply:
nice thanks this looks better!